DSpace/dspace-angular
# Fixed community's sub-communities and collections sort

## References
* Fixes #2160
* Requires DSpace/DSpace#8841

## Description
Fixed community's sub-communities and collections sort options from url being ignored when making the request to the backend (meaning that after refreshing the page your sort options would be lost). The pagination options for sub-communities were also lost after refreshing the page.

## Instructions for Reviewers
List of changes in this PR:
* Made changes to the `CommunityPageSubCollectionListComponent` & `CommunityPageSubCommunityListComponent` to retrieve the pagination options and the sort options from the url when available.

**Guidance for how to test this PR:**
- After testing the backend PR you should have a community containing at least 2 sub-communities and 2 collections with different names and go to the parent community page
- Click on the gear next to the collections and to the sub-communities and select `Descending`
- Check that the collections and the sub-communities are sorted in descending order
- Check that after refreshing the page the collections and the sub-communities are still sorted in descending order
- Click on the gear next to the collections and to the sub-communities and select `Ascending`
- Check that the collections and the sub-communities are sorted in ascending order
- Check that after refreshing the page the collections and the sub-communities are still sorted in ascending order
- Set the page size for the sub-communities to 1
- Check that after refreshing the page the page size is still 1

## Checklist
